trunner/runners/flasher.py

```diff
@@ -0,0 +1,108 @@
+#
+# Phoenix-RTOS test runner
+#
+# Flasher classes for device runners
+#
+# Copyright 2021, 2022 Phoenix Systems
+# Authors: Jakub Sarzyński, Damian Loewnau
+#
+
+from abc import ABC, abstractmethod
+import logging
+import sys
+
+from pexpect.exceptions import TIMEOUT, EOF
+from .common import PloError, PloTalker, RebootError
+from .wrappers import Psu, Gdb, Phoenixd, PhoenixdError, GdbError, append_output
+
+
+class Flasher(ABC):
+    """Common interface for flashing the system image"""
+
+    def __init__(self, target, serial_port, phoenixd_port, flash_memory, runner_reboot, logpath):
+        self.target = target
+        self.serial_port = serial_port
+        self.phoenixd_port = phoenixd_port
+        self.flash_memory = flash_memory
+        self.runner_reboot = runner_reboot
+        self.logpath = logpath
+
+    def flash(self):
+        phd = None
+        try:
+            self.reboot(flashing_mode=True)
+            with PloTalker(self.serial_port) as plo:
+                if self.logpath:
+                    plo.plo.logfile = open(self.logpath, "a")
```

Comment:
I know that this not change from this PR, but this should be placed in the PloTalker class method. Leave as is anyway, just marking for the future